tests/test_cwl.py

```diff
@@ -85,6 +85,61 @@ def test_subscription_filter(self):
         self.assertEqual(resources[0]["c7n:SubscriptionFilters"][0]["destinationArn"],
             "arn:aws:lambda:us-east-2:1111111111111:function:CloudCustodian")
 
+    def test_put_subscription_filter(self):
+        factory = self.replay_flight_data("test_log_group_put_subscription_filter")
+        log_group = "c7n-test-a"
+        filter_name = "log-susbscription-filter-a"
+        filter_pattern = "id"
+        destination_arn = "arn:aws:logs:us-east-1:644160558196:destination:lambda"
+        distribution = "ByLogStream"
+        client = factory().client("logs")
+        p = self.load_policy(
+            {
+                "name": "put-subscription-filter",
+                "resource": "log-group",
+                "filters": [{"logGroupName": log_group}],
+                "actions": [{"type": "put-subscription-filter",
+                "filter_name": filter_name,
+                "filter_pattern": filter_pattern,
+                "destination_arn": destination_arn}],
+            },
+            session_factory=factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(
+            client.describe_subscription_filters(logGroupName=log_group,
```

Comment:
@raghavendarreddyg Seems redundant to call `client.describe_subscription_filters()` multiple times. Can you just call once and save the result in a variable?

Reply:
@darrendao thanks for catching that. Modified as per your suggestion.